### PR TITLE
Fix Issue 20334 - posix.mak clean target does not remove all generated files

### PR DESCRIPTION
**Pull request by alexandrumc**
When the "runnable" test suite is run, it generates some files that are currently omitted by the existing Makefile clean target.

The fix was to extend the existing clean target with some additional remove commands. For the .o files, I added a "rm" command right in the script where they are generated.

### CODE REVIEW

**Review by RazvanN7 — APPROVED** [review]

**Review by atilaneves — APPROVED** [review]

**Comment by atilaneves** [comment]
I approved, but for the life of me I don't think I'll understand what the purpose of `make clean` is in a git world. `git clean -xffd` will always work. AFAIC this is a holdover from ye olden times of CVS.

**Comment by RazvanN7** [comment]
`git clean -xffd` might not be as obvious as `make clean`

**Comment by atilaneves** [comment]
> `git clean -xffd` might not be as obvious as `make clean`

Which makes me think the makefile should be changed to:

```
clean:
	git clean -xffd
```

File: test/runnable/mixin.sh [hunk 1]
```diff
@@ -9,3 +9,4 @@ tr '\\' '/' < ${tmp_file} > ${tmp_file}2
 diff -pu --strip-trailing-cr ${expect_file} ${tmp_file}2
 
 rm ${tmp_file} ${tmp_file}2
+rm ${TEST_NAME}.o
```
Comment (wilzbach) [inline test/runnable/mixin.sh]: Why not simply compile with `-o-`?

File: test/runnable/test_switches.sh [hunk 2]
```diff
@@ -18,6 +18,7 @@ src_file=${OUTPUT_BASE}/src.d
 clean()
 {
     rm -rf ${OUTPUT_BASE}
+    rm -rf mymod.o
```
Comment (wilzbach) [inline test/runnable/test_switches.sh]: Again, adding `-o-` would be the better solution.

**Review by wilzbach — DISMISSED** [review]

**Comment by MoonlightSentinel** [comment]
Not all untracked  files are build artifacts. 

E.g. small D files containing minimal examples

File: test/runnable/mixin.sh (hunk 1 above)
Comment (alexandrumc) [reply test/runnable/mixin.sh]: Yes, you are right.

**Comment by atilaneves** [comment]
> Not all untracked files are build artifacts.
> 
> E.g. small D files containing minimal examples

Which ones? If they're not in the repo, and they're not a build artifact... what are they?

**Comment by MoonlightSentinel** [comment]
> Which ones? If they're not in the repo, and they're not a build artifact... what are they?

I usually create a small example file when i want to debug dmd or reduce a test case (after dustmite). E.g. when working on `-checkaction=context` i had a small snippet like this:
```
void main() {
    assert( /* condition */ );
}
```

**Comment by MoonlightSentinel** [comment]
Walter also mentioned in #10453 that he sometimes creates copys of existing source files which probably shouldn't be deleted either

**Comment by atilaneves** [comment]
> > Which ones? If they're not in the repo, and they're not a build artifact... what are they?
> 
> I usually create a small example file when i want to debug dmd or reduce a test case (after dustmite). E.g. when working on `-checkaction=context` i had a small snippet like this:
> 
> ```
> void main() {
>     assert( /* condition */ );
> }
> ```

This way of working is alien to me. I put such files in /tmp. If I don't, *and* I want to keep them, I use `git stash`. If I forget, I go to my editor and retrieve the contents from RAM.

**Comment by MoonlightSentinel** [comment]
If that works for you its perfectly fine as workflow is always a matter of personal preference. I like my style because it integrates well with code-d and VS Codes Tasks but i wont claim that it is the perfect solution for everyone.

But to get back to the initial discussion, we should not change `make clean` to delete all untracked files as that would disrupt a lot of unsuspecting users.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**